lib/backend/k8s/k8s.go

```diff
@@ -739,9 +741,79 @@ func (c *KubeClient) deleteGlobalConfig(k *model.KVPair) error {
 }
 
 func (c *KubeClient) getHostConfig(k model.HostConfigKey) (*model.KVPair, error) {
+	if k.Name == "IpInIpTunnelAddr" {
+		n, err := c.clientSet.Nodes().Get(k.Hostname, metav1.GetOptions{})
+		if err != nil {
+			return nil, resources.K8sErrorToCalico(err, k)
+		}
+
+		kvp, err := getTunIp(n)
+		if err != nil {
+			return nil, err
+		}
+
+		return kvp, nil
+	}
+
 	return nil, errors.ErrorResourceDoesNotExist{Identifier: k}
 }
 
 func (c *KubeClient) listHostConfig(l model.HostConfigListOptions) ([]*model.KVPair, error) {
-	return []*model.KVPair{}, nil
+	var kvps = []*model.KVPair{}
+
+	// Short circuit if they aren't asking for information we can provide.
+	if l.Name == "" || l.Name == "IpInIpTunnelAddr" {
+		// First see if we were handed a specific host, if not list all Nodes
+		if l.Hostname == "" {
+			nodes, err := c.clientSet.Nodes().List(v1.ListOptions{})
+			if err != nil {
+				return nil, resources.K8sErrorToCalico(err, l)
+			}
+
+			for _, node := range nodes.Items {
+				kvp, err := getTunIp(&node)
+				if err != nil {
+					log.Warnf("Invalid IP for HostConfig: %s, %s", node.Name, node.Spec.PodCIDR)
+					continue
+				}
+
+				kvps = append(kvps, kvp)
+			}
+		} else {
+			node, err := c.clientSet.Nodes().Get(l.Hostname, metav1.GetOptions{})
+			if err != nil {
+				return nil, resources.K8sErrorToCalico(err, l)
+			}
+
+			kvp, err := getTunIp(node)
+			if err != nil {
+				return []*model.KVPair{}, nil
+			}
+
+			kvps = append(kvps, kvp)
+		}
+	}
+
+	return kvps, nil
 }
+
+func getTunIp(n *v1.Node) (*model.KVPair, error) {
+	ip, _, err := net.ParseCIDR(n.Spec.PodCIDR)
```

Comment:
As discussed - I think this should check if the CIDR exists and if so return no error and no KVPair - since it isn't really an error condition.  However, I think that means you'll also need to handle no KV returned in the above, which for a Get would correspond to a No such resource error.